contracts/osnft.sol

```diff
@@ -358,13 +342,42 @@ contract OSNFT is
         }
     }
 
+    // Owner Functions
+    function updateMintRoyalty(uint16 newMintRoyalty) external onlyOwner {
+        mintRoyalty = newMintRoyalty;
+        emit MintRoyaltyUpdated(newMintRoyalty);
+    }
+
+    function updateBurnRoyalty(uint16 newBurnRoyalty) external onlyOwner {
+        burnRoyalty = newBurnRoyalty;
+        emit BurnRoyaltyUpdated(newBurnRoyalty);
+    }
+
+    function withdrawEarning(
+        address tokenAddress,
+        uint256 amount
+    ) external onlyOwner {
+        withdrawEarningTo(owner(), tokenAddress, amount);
+    }
+
+    function withdrawEarningTo(
+        address accountTo,
+        address tokenAddress,
+        uint256 amount
+    ) public onlyOwner {
+        require(amount <= _earning[tokenAddress], "Amount exceed earning");
+        _earning[tokenAddress] -= amount;
+        _requirePaymentFromContract(tokenAddress, accountTo, amount);
+    }
+
+    // Overrides
     function _msgSender()
         internal
         view
         override(ContextUpgradeable, ERC2771ContextUpgradeable)
         returns (address sender)
     {
-        sender = ERC2771ContextUpgradeable._msgSender();
+        sender = super._msgSender();
```

Comment:
what was the issue with prev code ?

Reply:
In the future, if we inherit more contracts having _msgSender() to override, this will create a problem.